app/services/vacancies/import/sources/ark.rb

```diff
@@ -186,11 +187,13 @@ def organisation_fields(schools)
 
   def find_schools(item)
     multi_academy_trust = SchoolGroup.trusts.find_by(uid: TRUST_UID)
+
     school_urn = item.supp_value
+    schools = Organisation.where(urn: school_urn) if school_urn.present?
 
-    return [] if multi_academy_trust.blank? && school_urn.blank?
-    return Organisation.where(urn: school_urn) if multi_academy_trust.blank?
-    return Array(multi_academy_trust) if school_urn.blank?
+    return [] if multi_academy_trust.blank? && schools.blank?
+    return schools if multi_academy_trust.blank?
+    return Array(multi_academy_trust) if schools.blank?
```

Comment:
This is more of a refactor to avoid triggering DB queries unless the URN to search for is present.